### PR TITLE
Use parallel_tests

### PR DESCRIPTION
**Pull request by ekohl**

### CODE REVIEW

File: moduleroot/Rakefile [hunk 1]
```diff
@@ -32,4 +32,4 @@ KafoModuleLint::RakeTask.new do |config|
   config.pattern = <%= @configs['param_docs_pattern'].inspect %>
 end
 
-task :default => [:validate, :lint, :spec]
+task :default => [:validate, :lint, :parallel_spec]
```
Comment (domcleal) [inline moduleroot/Rakefile]: puppetlabs_spec_helper 2.0.0 has a `release_checks` task that calls validate, lint, parallel_spec, plus additional release-readiness tests which would be best to use here.

**Comment by ekohl** [comment]
Turns out parallel_tests needs at least ruby 2.0. I'm not sure how to do that in the Gemfile config.

File: moduleroot/Rakefile (hunk 1 above)
Comment (ekohl) [inline moduleroot/Rakefile]: Fixed

**Comment by ekohl** [comment]
Let's leave this open till we drop Puppet 3 so we no longer need to test on Ruby 1.9.3.

**Comment by bastelfreak** [comment]
looks fine to me, we have the same settings on Vox Pupuli

**Comment by ekohl** [comment]
Updated. It's slightly more complex due to Ruby 1.9.3 compatibility and uses release_checks only when parallel_tests is available.

File: config_defaults.yml [hunk 2]
```diff
@@ -9,7 +9,14 @@ Gemfile:
   - gem: rspec-puppet-facts
     version: '>= 1.7'
   - gem: puppetlabs_spec_helper
-    version: '>= 0.8.0'
+    version: ">= 1.1.0', '<= 2.1.0"
+  - gem: parallel_tests
+    options:
+      platforms:
```
Comment (ekohl) [inline config_defaults.yml]: So this part fails on older rubies. I haven't seen an option to do this on all buy ruby_19 platforms. Is there any?

Comment (bastelfreak) [inline config_defaults.yml]: do you think we still need to care about ruby1.9 (or even 2.0, 2.1)? They are all EOL since a long time.

Comment (mmoll) [inline config_defaults.yml]: we do.

Comment (ekohl) [inline config_defaults.yml]: For the 1.15 cycle we will continue support puppet 3 and on some platforms that runs on ruby 1.9. We will branch and pin there, so master will soon move to puppet 4 and drop it.

Comment (domcleal) [inline config_defaults.yml]: > So this part fails on older rubies. I haven't seen an option to do this on all buy ruby_19 platforms. Is there any?

You'd need to move it to the Gemfile proper, using `gem 'parallel_tests', [...] if RUBY_VERSION > '1.9'`.

Comment (ekohl) [inline config_defaults.yml]: Thanks. Tried to avoid modifying Gemfile directly but you're right. Updated

**Comment by ekohl** [comment]
Looks like puppet 3 + ruby 2+ fails now: https://travis-ci.org/theforeman/puppet-dns/builds/217143674
```
---> syntax:manifests
The use of 'import' is deprecated at 1. See http://links.puppetlabs.com/puppet-import-deprecation
   (at /home/travis/.rvm/gems/ruby-2.1.5/gems/puppet-3.8.7/lib/puppet/parser/parser_support.rb:110:in `import')

rake aborted!
/home/travis/.rvm/gems/ruby-2.1.5/gems/puppet-syntax-2.4.0/lib/puppet-syntax/tasks/puppet-syntax.rb:69:in `block (2 levels) in initialize'
/home/travis/.rvm/gems/ruby-2.1.5/gems/puppetlabs_spec_helper-2.1.0/lib/puppetlabs_spec_helper/rake_tasks.rb:468:in `block in <top (required)>'
/home/travis/.rvm/gems/ruby-2.1.5/gems/puppetlabs_spec_helper-2.1.0/lib/puppetlabs_spec_helper/rake_tasks.rb:517:in `block in <top (required)>'
/home/travis/.rvm/gems/ruby-2.1.5/bin/ruby_executable_hooks:15:in `eval'
/home/travis/.rvm/gems/ruby-2.1.5/bin/ruby_executable_hooks:15:in `<main>'
Tasks: TOP => syntax => syntax:manifests
(See full trace by running task with --trace)
```

**Comment by domcleal** [comment]
The release_checks task uses a different order to our existing tasks. It runs linting before validation, and kafo_parsers (via kafo_module_lint) triggers the deprecation warning on 3.x. puppet-syntax during validation later picks up the earlier deprecation warning. Validation before linting wouldn't trigger the same error.

**Comment by ekohl** [comment]
I opened https://github.com/theforeman/foreman-installer-modulesync/pull/57 to simply pin it until we drop Puppet 3.

**Comment by ekohl** [comment]
Updated in the simplified version. To be merged after https://github.com/theforeman/foreman-installer-modulesync/pull/58

**Pull request closed without merging**